utils/swift_build_support/swift_build_support/cmake.py

```diff
@@ -119,6 +119,8 @@ def common_options(self):
 
         if args.build_ninja and args.cmake_generator == 'Ninja':
             define('CMAKE_MAKE_PROGRAM', toolchain.ninja)
```

Comment:
``toolchain.ninja`` has the path to the newly built ninja.

If I remove the ninja src directory:
```
ergbook:swift erg$ ./utils/build-script -r -- --build-ninja
+ mkdir -p /Users/erg/src2/build/Ninja-RelWithDebInfoAssert
./utils/build-script: fatal error: can't find source directory for ninja (tried /Users/erg/src2/ninja)
```

If it's there:
```
ergbook:swift erg$ ./utils/build-script -r -- --build-ninja
('toolchain.ninja:', '/Users/erg/src2/build/Ninja-RelWithDebInfoAssert/ninja-build/ninja')
```

So it's not the most clear that ``ninja`` path will definitely be set, but it's the case.

What's a bit more troubling is that we set the ``build_ninja`` flag in python, even though the user did not request it, in the case that the cmake generator is ``None`` and we have no system-wide ninja installed. From ``build-script``:

```python
        # Set the default CMake generator.
        if args.cmake_generator is None:   # it's None unless you pass it on the command line
            args.cmake_generator = "Ninja"

        ninja_required = (           # this is True because 'Ninja' is set above
            args.cmake_generator == 'Ninja' or args.build_foundation)
        if ninja_required and toolchain.ninja is None:
            args.build_ninja = True   # If ninja is not installed on Linux, toolchain.ninja is None
                                                     # and we build it
```